scheduling/non_preemptive_shortest_job_first.py

```diff
@@ -0,0 +1,116 @@
+# SJF
+"""
+Non-preemptive Shortest Job First
+Shortest execution time process is chosen for the next execution.
+https://www.guru99.com/shortest-job-first-sjf-scheduling.html
+https://en.wikipedia.org/wiki/Shortest_job_next
+"""
+
+
+from __future__ import annotations
+
+from statistics import mean
+
+
+def calculate_waitingtime(
+    arrival_time: list[int], burst_time: list[int], no_of_processes: int
+) -> list[int]:
+
+    """
+    Calculate the waiting time of each processes
+
+    Return: The waiting time for each process.
+    >>> calculate_waitingtime([0,1,2], [10, 5, 8], 3)
+    [0, 9, 13]
+    >>> calculate_waitingtime([1,2,2,4], [4, 6, 3, 1], 4)
+    [0, 7, 4, 1]
+    >>> calculate_waitingtime([0,0,0], [12, 2, 10],3)
+    [12, 0, 2]
+    """
+
+    waiting_time = [0] * no_of_processes
+    remaining_time = [0] * no_of_processes
+
+    """
+    Initialize remaining_time to waiting_time.
+    """
+    for i in range(no_of_processes):
+        remaining_time[i] = burst_time[i]
+    ready_process = []
+
+    completed = 0
+    total_time = 0
+
+    """
+     When processes are not completed,
+        A process whose arrival time has passed \
+        and has remaining execution time is put into the ready_process.
+        The shortest process in the ready_process, target_process is executed.
+    """
+    while completed != no_of_processes:
```

Comment:
This block of code can be used to improve https://github.com/TheAlgorithms/Python/blob/master/scheduling/shortest_job_first.py